pkg/liqo-controller-manager/namespaceMap-controller/manage_remoteNamespaces.go

```diff
@@ -58,49 +59,58 @@ func (r *NamespaceMapReconciler) deleteRemoteNamespace(clusterID, remoteName str
 }
 
 // This function checks if there are Namespaces that have to be created or deleted, in accordance with DesiredMapping
-// and CurrentMapping fields.
-func (r *NamespaceMapReconciler) manageRemoteNamespaces(nm *mapsv1alpha1.NamespaceMap) error {
+// and CurrentMapping fields. It updates also NamespaceOffloading status in according to the remote Namespace Status.
+func (r *NamespaceMapReconciler) manageRemoteNamespaces(nm *mapsv1alpha1.NamespaceMap) (bool, error) {
+	requeue := false
 	if nm.Status.CurrentMapping == nil {
 		nm.Status.CurrentMapping = map[string]mapsv1alpha1.RemoteNamespaceStatus{}
 	}
 
 	// if DesiredMapping field has more entries than CurrentMapping, is necessary to create new remote namespaces
 	for localName, remoteName := range nm.Spec.DesiredMapping {
-		if _, ok := nm.Status.CurrentMapping[localName]; ok {
+		if remoteStatus, ok := nm.Status.CurrentMapping[localName]; ok && remoteStatus.Phase == mapsv1alpha1.MappingAccepted {
```

Comment:
May we at least cross-check that the namespace is still available and is labeled as expected?

Reply:
yes sure